pkg/controllers/dnspolicy/dns_helper.go

```diff
@@ -204,61 +282,27 @@ func (s *Service) SetEndpoints(ctx context.Context, mcgTarget *MultiClusterGatew
+				endpoint = createOrUpdateEndpoint(clusterLbName, ipValues, v1alpha1.ARecordType, "", dns.DefaultTTL, currentEndpoints)
+				newEndpoints = append(newEndpoints, endpoint)
 				hostValues = append(hostValues, clusterLbName)
 			}
 
 			for _, hostValue := range hostValues {
-				endpoint = createOrUpdateEndpoint(geoLbName, []string{hostValue}, v1alpha1.CNAMERecordType, hostValue, DefaultTTL, currentEndpoints)
-				endpoint.SetProviderSpecific(ProviderSpecificWeight, strconv.Itoa(cgwTarget.GetWeight()))
-				clusterEndpoints = append(clusterEndpoints, endpoint)
+				endpoint = createOrUpdateEndpoint(geoLbName, []string{hostValue}, v1alpha1.CNAMERecordType, hostValue, dns.DefaultTTL, currentEndpoints)
+				endpoint.SetProviderSpecific(dns.ProviderSpecificWeight, strconv.Itoa(cgwTarget.GetWeight()))
+				newEndpoints = append(newEndpoints, endpoint)
 			}
-		}
-		if len(clusterEndpoints) == 0 {
-			continue
-		}
-		newEndpoints = append(newEndpoints, clusterEndpoints...)
-
-		//Create lbName CNAME (lb-a1b2.shop.example.com -> default.lb-a1b2.shop.example.com)
-		endpoint = createOrUpdateEndpoint(lbName, []string{geoLbName}, v1alpha1.CNAMERecordType, string(geoCode), DefaultCnameTTL, currentEndpoints)
 
-		//Deal with the default geo endpoint first
-		if geoCode.IsDefaultCode() {
-			defaultEndpoint = endpoint
-			// continue here as we will add the `defaultEndpoint` later
-			continue
-		} else if (geoCode == mcgTarget.getDefaultGeo()) || defaultEndpoint == nil {
-			// Ensure that a `defaultEndpoint` is always set, but the expected default takes precedence
-			defaultEndpoint = createOrUpdateEndpoint(lbName, []string{geoLbName}, v1alpha1.CNAMERecordType, "default", DefaultCnameTTL, currentEndpoints)
-		}
-
-		if geoCode.IsContinentCode() {
-			endpoint.SetProviderSpecific(ProviderSpecificGeoContinentCode, string(geoCode))
-		} else if geoCode.IsCountryCode() {
-			endpoint.SetProviderSpecific(ProviderSpecificGeoCountryCode, string(geoCode))
 		}
-		newEndpoints = append(newEndpoints, endpoint)
 	}
 
-	if len(newEndpoints) > 0 {
-		// Add the `defaultEndpoint`, this should always be set by this point if `newEndpoints` isn't empty
-		defaultEndpoint.SetProviderSpecific(ProviderSpecificGeoCountryCode, "*")
-		newEndpoints = append(newEndpoints, defaultEndpoint)
-		//Create gwListenerHost CNAME (shop.example.com -> lb-a1b2.shop.example.com)
-		endpoint = createOrUpdateEndpoint(gwListenerHost, []string{lbName}, v1alpha1.CNAMERecordType, "", DefaultCnameTTL, currentEndpoints)
-		newEndpoints = append(newEndpoints, endpoint)
-	}
-
-	sort.Slice(newEndpoints, func(i, j int) bool {
-		return newEndpoints[i].SetID() < newEndpoints[j].SetID()
-	})
 	dnsRecord.Spec.Endpoints = newEndpoints
 
-	if !equality.Semantic.DeepEqual(old, dnsRecord) {
-		return s.controlClient.Update(ctx, dnsRecord)
+	if equality.Semantic.DeepEqual(old.Spec, dnsRecord.Spec) {
+		return nil
 	}
-	return nil
+
+	return dh.Update(ctx, dnsRecord, &client.UpdateOptions{})
 }
```

Comment:
This appears to be removing changes that were made on main. Is it up to date?

Reply:
will double check